internal/host/host.go

```diff
@@ -351,15 +351,30 @@ func (m *Manager) updateInventory(ctx context.Context, cluster *common.Cluster,
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
 
-	// Remove virtual interfaces if the feature is not enabled
-	if !m.Config.EnableVirtualInterfaces {
-		var physicalInterfaces []*models.Interface = make([]*models.Interface, 0)
-		for _, intf := range inventory.Interfaces {
-			if intf.Type == "" || intf.Type == "physical" || intf.Type == "vlan" || intf.Type == "bond" {
-				// Empty interface type indicates an older agent, which only passes physical interfaces
-				physicalInterfaces = append(physicalInterfaces, intf)
-			}
+	existingHostInventory, err := common.UnmarshalInventory(h.Inventory)
+	if err != nil {
+		log.WithError(err).Debugf("failed to unmarshal host existing inventory %s", h.Inventory)
+	}
+
+	var physicalInterfaces []*models.Interface = make([]*models.Interface, 0)
+	var virtualInterfaces []string = make([]string, 0)
+	for _, intf := range inventory.Interfaces {
+		if intf.Type == "" || intf.Type == "physical" || intf.Type == "vlan" || intf.Type == "bond" {
+			// Empty interface type indicates an older agent, which only passes physical interfaces
+			physicalInterfaces = append(physicalInterfaces, intf)
+		} else {
+			virtualInterfaces = append(virtualInterfaces, intf.Type)
 		}
+	}
+
+	// Only record metrics if it's new/changed inventory
+	if existingHostInventory == nil || (existingHostInventory != nil && len(physicalInterfaces) != len(existingHostInventory.Interfaces)) {
+		m.eventsHandler.AddMetricsEvent(ctx, *h.ClusterID, h.ID, models.EventSeverityInfo, "nic.virtual_interfaces", time.Now(), "count", len(virtualInterfaces), "types", virtualInterfaces)
```

Comment:
u should not add cluster/host id to the metrics. It will make it not really useful or queriable. 
And why do we actually need it per host/cluster. We just want to see how many of such are out there, not to use metrics to track down the cluster - for that we can query the inventory, dont we?

Reply:
@romfreiman this metric event API will add an event to DB and we will see it in the elastic - not in Prometheus. I don't think it's a problem to have cluster-id & host id if we are calling it only **once** for each host. 
@CrystalChun I think that we are calling updateInventory method many times so I don't think it's a good place to call this code.
Do we want to see this data only on clusters that started installation or on all clusters?

Reply:
@gamli75 thanks for clarifying. then please ignore my comment :) I saw metric in the name and in the pr title

Reply:
@gamli75 this [condition](https://github.com/openshift/assisted-service/pull/4079/files#diff-549fd6f951612ce99043912371fffacd86d8249a40ef3905825be3d45f81dba4R371) ensures that the metrics is only called if the interfaces are new or have changed. 

> Do we want to see this data only on clusters that started installation or on all clusters?

I think it would be useful to see it on all clusters, regardless if they started install or not. What do you all think? @gamli75 @avishayt @filanov

Reply:
I don't know if we will care about interfaces for clusters that didn't install, but does having the information cost us anything?

Reply:
it will cost us more events in the DB

Reply:
@CrystalChun can we run cluster installation with this code and verify the number of events?

Reply:
Assuming the number of interfaces doesn't change it should be 1 event per cluster

Reply:
@CrystalChun Please make sure this is covered by UT

Reply:
That's correct, it should be 1 event per cluster

@avishayt will do!

Reply:
If we want to skip clusters that didn't start the installation it will need to hold the information until the installation is starting, meaning that we will need to store it in the DB, the flow that we have right now we don't store extra data. 
It will make the solution much more complicated, having another event is a small price to pay.